### PR TITLE
[core] Renamed _max_retries in actor methods to max_task_retries.

### PR DESCRIPTION
**Pull request by rynewang**
Renamed _max_retries in actor methods to max_task_retries.

Also added tests for threaded actors

Signed-off-by: Ruiyang Wang <rywang014@gmail.com>

### CODE REVIEW

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
premerge failures

**Comment by rkooo567** [comment]
premerge failures

**Comment by rynewang** [comment]
premerge failures fixed (retried and passed)

**Comment by rkooo567** [comment]
Need doc approval

File: doc/source/ray-core/fault_tolerance/actors.rst
Comment (angelinalg) [inline doc/source/ray-core/fault_tolerance/actors.rst:153]:
```suggestion
``max_task_retries`` applies to both exceptions and actor crashes. A Ray actor can set this option to apply to all of its methods. A method can also set an overriding option for itself. Ray searches for the first non-default value of ``max_task_retries`` in this order:
```

File: doc/source/ray-core/fault_tolerance/actors.rst
Comment (angelinalg) [inline doc/source/ray-core/fault_tolerance/actors.rst:155]:
```suggestion
- The method call's value, for example, `actor.method.options(max_task_retries=2)`. Ray ignores this value if you don't set it.
```

File: doc/source/ray-core/fault_tolerance/actors.rst
Comment (angelinalg) [inline doc/source/ray-core/fault_tolerance/actors.rst:156]:
```suggestion
- The method definition's value, for example, `@ray.method(max_task_retries=2)`. Ray ignores this value if you don't set it.
```

**Review by angelinalg — APPROVED** [review]
Just a few minor nits.

**Comment by rkooo567** [comment]
lint failure (and premerge)

**Comment by rkooo567** [comment]
ping on lint failure!

**Comment by rynewang** [comment]
No more lint failure after merge master. Still a "ml: train gpu tests" failure which looks to be fail to build and irrelavent.

**Comment by rkooo567** [comment]
CAn't merge without passing premerge. Retrying...

**Comment by rynewang** [comment]
Ready to merge

**Pull request merged**
